### DISCUSSION TITLE
Concurrency Benchmark results

### DISCUSSION THREAD

**Discussion by bgervan** [emmett-framework/granian#354] [General]
Hey, 

I find a bit odd that the best results is with the single core single thread single process? Isn't the more process should be better on a multicore system? What am I missing here?

Thanks

**Comment by gi0baro** [comment]
I find it odd as well, but if you check the history seems to be quite different depending on the OS.

It seems on MacOS there's no such a difference, whether on Linux it kinda sits in the range N cores/2-N cores depending on the protocol. There might be several impacting factors for this, from uvloop to Tokio behaving differently on such OS to the CPU architecture, to the number of performance cores VS efficiency cores.

Also mind that I don't have a Linux machine with tons of CPU cores to run these benchmarks against, and GHA have just a few cores available on the free instances. I'm open to suggestions/PRs to improve the concurrency benchmarks.
